src/screens/ChannelListScreen/index.tsx

```diff
@@ -1,19 +1,37 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 /* eslint-disable no-use-before-define */
 import * as React from 'react';
-import {FlatList} from 'react-native';
+import {FlatList, StatusBar, View} from 'react-native';
+import {useNavigation} from '@react-navigation/core';
 
+import ChannelListHeaderItem from '../../components/ChatList/ChannelListHeaderItem';
 import CommunityChannelItem from '../../components/ChatList/CommunityChannelItem';
 import GroupChatChannelItem from '../../components/ChatList/GroupChatChannelItem';
 import MessageChannelItem from '../../components/AnonymousChat/MessageChannelItem';
 import PostNotificationChannelItem from '../../components/AnonymousChat/PostNotificationChannelItem';
+import Search from './elements/Search';
 import useFollowUser from './hooks/useFollowUser';
+import useLocalDatabaseHook from '../../database/hooks/useLocalDatabaseHook';
+import useRootChannelListHook from '../../hooks/screen/useRootChannelListHook';
 import useSignedChannelListScreenHook from '../../hooks/screen/useSignedChannelListHook';
+import useUserAuthHook from '../../hooks/core/auth/useUserAuthHook';
 
-const ChannelListScreen = () => {
-  const {channels, goToChatScreen, goToPostDetailScreen, goToCommunityScreen} =
-    useSignedChannelListScreenHook();
+const ChannelListScreen = ({route}) => {
+  const {refresh} = useLocalDatabaseHook();
+  const {checkNotificationPermission} = useRootChannelListHook();
   const {handleFollow, isInitialFollowing} = useFollowUser();
+  const {profile} = useUserAuthHook();
+  const navigation = useNavigation();
+  const isFocused = navigation.isFocused();
+  const {channels, goToChatScreen, goToPostDetailScreen, goToCommunityScreen, goToContactScreen} =
+    useSignedChannelListScreenHook();
+
+  React.useEffect(() => {
+    if (isFocused) {
+      refresh('channelList');
+      checkNotificationPermission();
+    }
+  }, [isFocused]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code introduces several hooks and components that were not present in the old code. It's good to see the use of hooks for managing state and side effects, but it's important to ensure that these hooks are used correctly. 

In line 19, `useLocalDatabaseHook` is used to get a `refresh` function, which is then called inside a `React.useEffect` hook whenever `isFocused` changes. This seems like a reasonable way to refresh the channel list when the screen comes into focus.

However, there's no error handling around the `refresh` call. If this function throws an error (for example, if there's a problem with the database), it will cause the component to unmount. To improve error handling, you could wrap the `refresh` call in a try-catch block and handle any errors appropriately.

```diff
  React.useEffect(() => {
    if (isFocused) {
      try {
        refresh('channelList');
      } catch (error) {
        console.error(error);
      }
      checkNotificationPermission();
    }
  }, [isFocused]);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->